forest/jasmine/traj2stats.py

```diff
@@ -140,175 +147,649 @@ def get_nearby_locations(traj: np.ndarray) -> Tuple[dict, dict, dict]:
     return ids, locations, tags
 
 
-def gps_summaries(traj,tz_str,option):
-    """
-    This function derives summary statistics from the imputed trajectories
-    if the option is hourly, it returns ["year","month","day","hour","obs_duration","pause_time","flight_time","home_time",
-    "max_dist_home", "dist_traveled","av_flight_length","sd_flight_length","av_flight_duration","sd_flight_duration"]
-    if the option is daily, it additionally returns ["obs_day","obs_night","radius","diameter","num_sig_places","entropy"]
-    Args: traj: 2d array, output from Imp2traj(), which is a n by 8 mat, with headers as [s,x0,y0,t0,x1,y1,t1,obs]
-          where s means status (1 as flight and 0 as pause), x0,y0,t0: starting lat,lon,timestamp,
-          x1,y1,t1: ending lat,lon,timestamp,  obs (1 as observed and 0 as imputed)
-          tz_str: timezone
-          option: 'daily' or 'hourly'
-    Return: a pd dataframe, with each row as an hour/day, and each col as a feature/stat
+def gps_summaries(
+    traj: np.ndarray,
+    tz_str: str,
+    frequency: Frequency,
+    places_of_interest: Union[List[str], None] = None,
+    save_log: bool = False,
+    threshold: Union[int, None] = None,
+    split_day_night: bool = False,
+    person_point_radius: float = 2,
+    place_point_radius: float = 7.5,
+) -> Tuple[pd.DataFrame, dict]:
+    """This function derives summary statistics from the imputed trajectories
+
+    If the option is hourly, it returns
+    ["year","month","day","hour","obs_duration","pause_time","flight_time","home_time",
+    "max_dist_home", "dist_traveled","av_flight_length","sd_flight_length",
+    "av_flight_duration","sd_flight_duration"]
+    if the option is daily, it additionally returns
+    ["obs_day","obs_night","radius","diameter","num_sig_places","entropy"]
+
+    Args:
+        traj: 2d array, output from Imp2traj(), which is a n by 8 mat,
+            with headers as [s,x0,y0,t0,x1,y1,t1,obs]
+            where s means status (1 as flight and 0 as pause),
+            x0,y0,t0: starting lat,lon,timestamp,
+            x1,y1,t1: ending lat,lon,timestamp,
+            obs (1 as observed and 0 as imputed)
+        tz_str: timezone
+        option: Frequency class
```

Comment:
Should be something like `frequency: how often summary statistics are calculated`?